edge/pkg/devicetwin/dmiclient/client.go

```diff
@@ -0,0 +1,75 @@
+package dmiclient
+
+import (
+	"net"
+	"time"
+
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+	"k8s.io/klog/v2"
+
+	pb "github.com/kubeedge/kubeedge/edge/pkg/apis/dmi/v1"
+	"github.com/kubeedge/kubeedge/edge/pkg/devicetwin/dtcommon"
+	"github.com/kubeedge/kubeedge/pkg/apis/devices/v1alpha2"
+)
+
+const (
+	unix string = "unix"
+)
+
+func GenerateDMIClient(sockPath string) (pb.DeviceMapperServiceClient, context.Context, *grpc.ClientConn, context.CancelFunc, error) {
+	dialer := func(addr string, t time.Duration) (net.Conn, error) {
+		return net.Dial(unix, addr)
+	}
+
+	conn, err := grpc.Dial(sockPath, grpc.WithInsecure(), grpc.WithDialer(dialer))
+	if err != nil {
+		klog.Errorf("did not connect: %v\n", err)
+		return nil, nil, nil, nil, err
+	}
+
+	c := pb.NewMapperClient(conn)
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
```

Comment:
why return context.Context context.CancelFunc in this func and it is not used in the func

Reply:
I want to make the code more uniform and less repetitive. if i move the ctx and cancelFunc outside, i will repeat this part of code every time when i generate a DMI-Client.

Reply:
i have re-write the part of dmi-client. this problem does not exist.